File: Editors/DisabledEditorsColor/disablededitorsvb-zip/DisabledEditorsVB/DisabledEditors/MyRadDateTimePicker.vb
Imports System.Text
Imports Telerik.WinControls.UI

Namespace DisabledEditors
	Public Class MyRadDateTimePicker
		Inherits RadDateTimePicker
       
        Private mtb As MyTextBox
		Private tbi As RadTextBoxItem

		Protected Overrides Sub CreateChildItems(ByVal parent As Telerik.WinControls.RadElement)
			MyBase.CreateChildItems(parent)

            mtb = New MyTextBox()
			tbi = New RadTextBoxItem(mtb)
			mtb.Font = Me.Font
			Me.DateTimePickerElement.TextBoxElement.Children.Add(tbi)
		End Sub

		Protected Overrides Sub OnEnabledChanged(ByVal e As EventArgs)
			If Not Me.Enabled Then
				tbi.Text = Me.DateTimePickerElement.TextBoxElement.TextBoxItem.Text
				Me.DateTimePickerElement.TextBoxElement.TextBoxItem.Visibility = Telerik.WinControls.ElementVisibility.Hidden
			Else
				Me.DateTimePickerElement.TextBoxElement.TextBoxItem.Visibility = Telerik.WinControls.ElementVisibility.Visible
			End If

			MyBase.OnEnabledChanged(e)
        End Sub

        Public Overrides Property ThemeClassName() As String
            Get
                Return GetType(RadDateTimePicker).FullName
            End Get
            Set(ByVal value As String)

            End Set
        End Property
	End Class
End Namespace

File: Editors/DisabledEditorsColor/disablededitorscs-zip/DisabledEditorsCS/DisabledEditors/MyRadDateTimePicker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;

namespace DisabledEditors
{
    public class MyRadDateTimePicker : RadDateTimePicker
    {
        public override string ThemeClassName
        {
            get
            {
                return typeof(RadDateTimePicker).FullName;
            }
        }

        MyTextBox mtb = new MyTextBox();
        RadTextBoxItem tbi;

        protected override void CreateChildItems(Telerik.WinControls.RadElement parent)
        {
            base.CreateChildItems(parent);

            tbi = new RadTextBoxItem(mtb);
            mtb.Font = this.Font;
            this.DateTimePickerElement.TextBoxElement.Children.Add(tbi);
        }

        protected override void OnEnabledChanged(EventArgs e)
        {
            if (!this.Enabled)
            {
                tbi.Text = this.DateTimePickerElement.TextBoxElement.TextBoxItem.Text;
                this.DateTimePickerElement.TextBoxElement.TextBoxItem.Visibility = Telerik.WinControls.ElementVisibility.Hidden;
            }
            else
            {
                this.DateTimePickerElement.TextBoxElement.TextBoxItem.Visibility = Telerik.WinControls.ElementVisibility.Visible;
            }

            base.OnEnabledChanged(e);
        }
    }
}
